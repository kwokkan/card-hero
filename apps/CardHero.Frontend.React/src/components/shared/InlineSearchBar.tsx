import React, { useState, useEffect } from "react";

interface IInlineSearchBarProps {
    value?: string;
    onValueUpdated?: (value?: string) => void;
}

export function InlineSearchBar(props: IInlineSearchBarProps) {
    const [value, setValue] = useState(props.value);

    useEffect(() => {
        onValueUpdated();
    }, [value]);

    const onValueUpdated = () => {
        if (props.onValueUpdated) {
            props.onValueUpdated(value);
        }
    };

    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setValue(event.currentTarget.value);
    };

    const onKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
        if (event.key === "Enter") {
            setValue(event.currentTarget.value);
        }
    };

    return (
        <div className="input-group">
            <input type="text" className="form-control" placeholder="Search" value={props.value} onChange={onChange} onKeyUp={onKeyUp} />
            <div className="input-group-append">
                <button type="button" className="btn btn-primary" onClick={onValueUpdated}>Search</button>
            </div>
        </div>
    );
}
