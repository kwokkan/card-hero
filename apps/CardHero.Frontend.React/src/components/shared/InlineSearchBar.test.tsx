import React from "react";
import renderer from "react-test-renderer";
import { InlineSearchBar } from "./InlineSearchBar";

test("<InlineSearchBar /> renders without default value", () => {
    const tree = renderer
        .create(
            <InlineSearchBar />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<InlineSearchBar /> renders with default value", () => {
    const tree = renderer
        .create(
            <InlineSearchBar value="Test" />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
