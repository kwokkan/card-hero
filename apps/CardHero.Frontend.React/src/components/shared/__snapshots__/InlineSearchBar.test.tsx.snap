// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineSearchBar /> renders with default value 1`] = `
<div
  className="input-group"
>
  <input
    className="form-control"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search"
    type="text"
    value="Test"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;

exports[`<InlineSearchBar /> renders without default value 1`] = `
<div
  className="input-group"
>
  <input
    className="form-control"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search"
    type="text"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
